libraries/AP_HAL_Linux/Scheduler.cpp

```diff
@@ -84,17 +84,17 @@ void Scheduler::init()
 
     mlockall(MCL_CURRENT|MCL_FUTURE);
 
-    if (geteuid() != 0) {
+    if (geteuid() == 0) {
+        struct sched_param param = { .sched_priority = APM_LINUX_MAIN_PRIORITY };
+        ret = sched_setscheduler(0, SCHED_FIFO, &param);
+        if (ret == -1) {
+            AP_HAL::panic("Scheduler: failed to set scheduling parameters: %s",
+                          strerror(errno));
+        }
+    } else {
         printf("WARNING: running as non-root. Will not use realtime scheduling\n");
     }
 
-    struct sched_param param = { .sched_priority = APM_LINUX_MAIN_PRIORITY };
-    ret = sched_setscheduler(0, SCHED_FIFO, &param);
-    if (ret == -1) {
-        AP_HAL::panic("Scheduler: failed to set scheduling parameters: %s",
-                      strerror(errno));
-    }
-
```

Comment:
what about all the other threads below?

Also, checking if we are root and just printing a warning is really bad. I don't want people running the flight stack without RT scheduling and then complaining about schedule overruns.

Could we add an argument to the init method to tell if we want RT scheduling? Then set it on Replay and nothing else.